### DISCUSSION TITLE
Set engine globally in website project

### DISCUSSION THREAD

**Discussion by cbrnr** [quarto-dev/quarto-cli#2498] [Q&A]
I have a website project and I'm trying to set the default engine to `knitr`. I tried adding this to my `_quarto.yml`:

```
engine:
  knitr
```

However, this doesn't seem to have any effect, because all of my sub-documents still use the Jupyter engine.

**Comment by dragonstyle** [comment]
I would expect this to work:

```
engine: knitr
```

Note that the option is on the same line (the indented next line actually makes that a sub-property of `engine` rather than it's value).

More on engine binding here:

https://quarto.org/docs/computations/execution-options.html#engine-binding

**Reply by cbrnr** [reply]
Thanks, this works!

**Comment by cscheid** [comment]
That's a brutally bad YAML interpretation though on our part, though! Let's see if I can improve it.

**Comment by jamesowers-cohere** [comment]
So this _doesn't_ work for me. If I have a ```{bash} cell in a `.qmd` file, quarto tries to use jupyter, even though I have the global setting in my _quarto.yml. Has something changed?

**Reply by mcanouil** [reply]
Nothing changed. There are no global engine specifications.

- https://github.com/quarto-dev/quarto-cli/issues/3157

(This issue was raised less than two months after this thread)
